### PR TITLE
Stand with Ukraine

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | n/a 
| License       | MIT
| Doc PR        | n/a

On 4.4
image

On 5.4
image

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]

**Review by fancyweb — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php [hunk 1]
```diff
@@ -148,7 +148,7 @@ public function all($namespace = null)
      */
     public function getLongVersion()
     {
-        return parent::getLongVersion().sprintf(' (env: <comment>%s</>, debug: <comment>%s</>)', $this->kernel->getEnvironment(), $this->kernel->isDebug() ? 'true' : 'false');
+        return parent::getLongVersion().sprintf(' (env: <comment>%s</>, debug: <comment>%s</>) <bg=#0057B7;fg=#FFDD00>#StandWith</><bg=#FFDD00;fg=#0057B7>Ukraine</> <href=https://sf.to/ukraine>Support Ukraine</>', $this->kernel->getEnvironment(), $this->kernel->isDebug() ? 'true' : 'false');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Console/Application.php]:
```suggestion
        return parent::getLongVersion().sprintf(' (env: <comment>%s</>, debug: <comment>%s</>) <bg=#0057B7;fg=#FFDD00;href=https://sf.to/ukraine>#StandWith</><bg=#FFDD00;fg=#0057B7;href=https://sf.to/ukraine>Ukraine</>', $this->kernel->getEnvironment(), $this->kernel->isDebug() ? 'true' : 'false');
```

**Review by nicolas-grekas — APPROVED** [review]
:+1: with one suggestion and one test to fix before merging. Thanks!

File: src/Symfony/Bundle/FrameworkBundle/Console/Application.php (hunk 1 above)
Comment (javiereguiluz) [reply src/Symfony/Bundle/FrameworkBundle/Console/Application.php]: I'm not sure about merging the banner and the link. Terminals display links in a very subtle way, so maybe it's not as visible as showing the banner and the link separately.

**Comment by AsuRaHan** [comment]
Слава Украине!

**Review by kbond — APPROVED** [review]

**Comment by yunmoxue** [comment]
It's a bad pr. It breaks the development experience.
why did it on this?

**Comment by derrabus** [comment]
> It breaks the development experience.

Please elaborate.

**Comment by kevinpapst** [comment]
I wanted to check how the result of this PR looks like in my SF 4.4 application (Console v4.4.37) and it seems the OutputFormatterStyle does not allow arbitrary colors. 

PHPStorm Console (Mac, zsh):
Bildschirmfoto 2022-03-08 um 14 38 10

Replacing the colors with `yellow` and `blue` would work:
Bildschirmfoto 2022-03-08 um 14 49 47

**Comment by yunmoxue** [comment]
> > It breaks the development experience.
> 
> Please elaborate.

@derrabus 
Code is pure, clean, simple.  There should be no political positions.
There is one time, will be countless times.
Today it could be Ukraine, tomorrow could be CN, US, GB or other.

I don't mean to be rude, but seeing “stand with Ukraine” will make me unhappy, beacause the symfony is the greate framework in my heart, i don’t want it to be a political tool.

but sadly, and your ceo looks very glad to do it.
look https://github.com/symfony-cli/symfony-cli/commit/d09ad2d8eb57a9bc910226a188bea3c8d240ffca

**Comment by fabpot** [comment]
My mistake, I created the PR on 5.4 first, then backported it to 4.4. Will fix it before the merge of course.

**Review by cesarkohl — APPROVED** [review]

**Comment by williamdes** [comment]
> > > It breaks the development experience.
> > 
> > 
> > Please elaborate.
> 
> @derrabus Code is pure, clean, simple. There should be no political positions. There is one time, will be countless times. Today it could be Ukraine, tomorrow could be CN, US, GB or other.
> 
> I don't mean to be rude, but seeing “stand with Ukraine” will make me unhappy, beacause the symfony is the greate framework in my heart, i don’t want it to be a political tool.
> 
> but sadly, and your ceo looks very glad to do it. look [symfony-cli/symfony-cli@d09ad2d](https://github.com/symfony-cli/symfony-cli/commit/d09ad2d8eb57a9bc910226a188bea3c8d240ffca)

I agree too, in my opinion code should not spread any political position.
That said if tools want to add some notice in places that do not affect any user directly (like packagist or phpunit did in the version lines) that's kind of okay.

**Comment by fabpot** [comment]
> That said if tools want to add some notice in places that do not affect any user directly (like packagist or phpunit did in the version lines) that's kind of okay.

Which is exactly what we are doing here, right?

**Comment by yunmoxue** [comment]
> > That said if tools want to add some notice in places that do not affect any user directly (like packagist or phpunit did in the version lines) that's kind of okay.
> 
> Which is exactly what we are doing here, right?

image

I think you put your idea on the website is enough.  There is no need to add it to the code.

even It's already affecting me to read the document, and not only me.

**Comment by williamdes** [comment]
> > That said if tools want to add some notice in places that do not affect any user directly (like packagist or phpunit did in the version lines) that's kind of okay.
> 
> Which is exactly what we are doing here, right?

Probably, this PR is missing a description screenshot as reading the code is not very clear how it will affect outputs ;)

**Comment by derrabus** [comment]
> There should be no political positions.

Opposing a war is not a political position. Helping the victims of a war is not a political position.

**Comment by fabpot** [comment]
Screenshots were added for more clarity.

**Review by williamdes — APPROVED** [review]

**Comment by yunmoxue** [comment]
> > There should be no political positions.
> 
> Opposing a war is not a political position. Helping the victims of a war is not a political position.

yes, i absolutely agree with you.

Uh....em...  but, is not The Syrian and Iraq in the war? 

i will never reply it, If you put the text "love and peace", "hope the world haven't war".

you stand with Ukraine , why don't stand with Syrian and Iraq or other country in the war together?

it's is why i say it is political position.

**Comment by derrabus** [comment]
> Uh....em... but, is not The Syrian and Iraq in the war?

Please look up "[whataboutism](https://en.wikipedia.org/wiki/Whataboutism)". I'm not taking that bait.

**Comment by nicolas-grekas** [comment]
I think open-source (and thus Symfony) is a by-product of global peace.
The war in Ukraine is a direct threat to global peace. This is what makes the situation there unique IMHO.
Being politically neutral doesn't mean not fighting for your roots when they are in danger.
And OSS is one of Symfony's root (just one very narrow example of those roots btw.)

:+1: again on my side.

**Comment by yunmoxue** [comment]
> > Uh....em... but, is not The Syrian and Iraq in the war?
> 
> Please look up "whataboutism". I'm not taking that bait.

i searched and get the message.

"On February 24, there were 4 air strikes around the world: 1. The Israeli Air Force attacked the Syrian army. 2. The Saudi Air Force airstrikes Yemeni civilians. 3. The United States bombed Somalia. 4. Russia launches airstrikes on Ukrainian military installations. 4 air raids, the first 3 are little known, the 4th is well known;"

i don't stand any political position.
i don't think Russia is right.
i hate the war.
and i care the Ukraine people and any people under the war include the Russia civilian.

i just speaking my viewpoint on the code. 

"don't put 'stand any country' on the code"

you can ignore if you think it is "bait".

**Comment by yoander** [comment]
> > There should be no political positions.
> 
> Opposing a war is not a political position. Helping the victims of a war is not a political position.

Of course it is not political position but I don't see any similar actions for Palestine or Libya (2011) Also I propose to StandWithAfrica only 10% fully vacinnated against COVID-19.

**Comment by yoander** [comment]
> Please look up "[whataboutism](https://en.wikipedia.org/wiki/Whataboutism)". I'm not taking that bait.

Yes you're right, you're the only one who's right!

**Pull request merged**